# 6.1 Sourcing Open Data: Airbnb Mallorca

## Table of Contents
### 1. Importing libraries and dataset
### 2. Data Wrangling
### 3. Data Cleaning
### 4. Data Understanding

### 1. Importing libraries and dataset

In [1]:
# import libraries 
import pandas as pd
import numpy as np
import os

In [2]:
# create path
path = r'C:\Users\svand\Documents\15-04-2024 Airbnb Mallorca'

In [3]:
# import Mallorca listings dataset
df_listings = pd.read_csv(os.path.join(path, '02 Data', 'Original Data', 'listings_mallorca.csv'), index_col = False)

In [4]:
df_listings.head()

Unnamed: 0                   id  \
0           0             18184333   
1           3   664248507739787372   
2           4   629002706694235276   
3           7  1006393514540791109   
4          11             41395364   

                                        listing_url       scrape_id  \
0             https://www.airbnb.com/rooms/18184333  20240323015309   
1   https://www.airbnb.com/rooms/664248507739787372  20240323015309   
2   https://www.airbnb.com/rooms/629002706694235276  20240323015309   
3  https://www.airbnb.com/rooms/1006393514540791109  20240323015309   
4             https://www.airbnb.com/rooms/41395364  20240323015309   

  last_scraped       source                                              name  \
0   2024-03-23  city scrape                                    Quinze Germans   
1   2024-03-23  city scrape                                Villa Alcanada Mar   
2   2024-03-23  city scrape  Villa Beach Alcudia, near the beach with jacuzzi   
3   2024-03-23  city scrape                    Habitación matrimonial con A/C   
4   2024-03-23  city scrape                          SANTA CATALINA PENTHOUSE   

                                         description  \
0  This modern terraced house in Puerto de Andrat...   
1  Villa Alcanada Mar is a superb recent addition...   
2  Relax with the whole family!<br />A few meters...   
3  Forget the worries at this great home: You wil...   
4  Stunning penthouse in the well-known Santa Cat...   

                               neighborhood_overview  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  Conocido barrio de Santa Catalina, con paseos,...   

                                         picture_url  ...  \
0  https://a0.muscache.com/pictures/prohost-api/H...  ...   
1  https://a0.muscache.com/pictures/prohost-api/H...  ...   
2  https://a0.muscache.com/pictures/miso/Hosting-...  ...   
3  https://a0.muscache.com/pictures/miso/Hosting-...  ...   
4  https://a0.muscache.com/pictures/8bda3e0a-3dd7...  ...   

   review_scores_communication review_scores_location review_scores_value  \
0                         4.78                   4.94                4.28   
1                         4.00                   5.00                5.00   
2                         4.90                   5.00                5.00   
3                         5.00                   5.00                4.00   
4                         4.75                   4.77                4.50   

     license instant_bookable calculated_host_listings_count  \
0  ETV/10044                t                            824   
1    ET/2290                f                             39   
2  1498/2018                f                              9   
3        NaN                f                              2   
4        NaN                t                              2   

  calculated_host_listings_count_entire_homes  \
0                                         821   
1                                          39   
2                                           9   
3                                           0   
4                                           2   

  calculated_host_listings_count_private_rooms  \
0                                            3   
1                                            0   
2                                            0   
3                                            2   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              0.23  
1                                           0              0.09  
2                                           0              0.47  
3                                           0              0.77  
4                                           0

In [5]:
df_listings.shape

(11987, 76)

### 2. Data Cleaning

##### In Excel I have looked at the dataset and made a selection of which columns I keep and which ones I can remove for my project. I will first make a dataframe of the columns I keep before I do consistency checks and data cleaning.

In [6]:
columns_to_keep = ['id', 'listing_url', 'name', 'picture_url', 'host_since', 'host_is_superhost', 'neighbourhood_cleansed', 'latitude', 'longitude', 'property_type', 'accommodates', 'bedrooms', 'amenities', 'price', 'minimum_nights', 'maximum_nights', 'has_availability', 'availability_30', 'availability_60', 'availability_90', 'availability_365', 'number_of_reviews', 'number_of_reviews_ltm', 'number_of_reviews_l30d', 'first_review', 'last_review', 'review_scores_rating', 'review_scores_accuracy', 'review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 'review_scores_location', 'review_scores_value', 'reviews_per_month']
df_selected = df_listings[columns_to_keep]

In [7]:
df_selected.shape

(11987, 34)

In [8]:
# check for mixed types
for col in df_selected.columns.tolist():
  weird = (df_selected[[col]].applymap(type) != df_selected[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df_selected[weird]) > 0:
    print (col)

host_is_superhost
has_availability


C:\Users\svand\AppData\Local\Temp\ipykernel_10624\795094372.py:3: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df_selected[[col]].applymap(type) != df_selected[[col]].iloc[0].apply(type)).any(axis = 1)
C:\Users\svand\AppData\Local\Temp\ipykernel_10624\795094372.py:3: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df_selected[[col]].applymap(type) != df_selected[[col]].iloc[0].apply(type)).any(axis = 1)
C:\Users\svand\AppData\Local\Temp\ipykernel_10624\795094372.py:3: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df_selected[[col]].applymap(type) != df_selected[[col]].iloc[0].apply(type)).any(axis = 1)
C:\Users\svand\AppData\Local\Temp\ipykernel_10624\795094372.py:3: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df_selected[[col]].applymap(type) != df_selected[[col]].iloc[0].apply(type)).any(axis = 1)


In [11]:
df_selected['host_is_superhost'].value_counts(dropna = False)

host_is_superhost
f      8774
t      3208
NaN       5
Name: count, dtype: int64

In [12]:
df_selected['has_availability'].value_counts(dropna = False)

has_availability
t      11983
f          3
NaN        1
Name: count, dtype: int64

##### Both columns are true/false columns with missing values. This is why it shows as mixed-type columns. Given the fact that in total there are only 6 rows with missing values for airbnb listings I am going to remove those.

In [13]:
# finding missing values
df_selected.isnull().sum()

id                             0
listing_url                    0
name                           0
picture_url                    0
host_since                     0
host_is_superhost              5
neighbourhood_cleansed         0
latitude                       0
longitude                      0
property_type                  0
accommodates                   0
bedrooms                       1
amenities                      0
price                          0
minimum_nights                 0
maximum_nights                 0
has_availability               1
availability_30                0
availability_60                0
availability_90                0
availability_365               0
number_of_reviews              0
number_of_reviews_ltm          0
number_of_reviews_l30d         0
first_review                   0
last_review                    0
review_scores_rating           0
review_scores_accuracy         4
review_scores_cleanliness      5
review_scores_checkin          4
review_sco

In [14]:
df_selected.dropna(subset = ['host_is_superhost', 'has_availability'], inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\1026239031.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.dropna(subset = ['host_is_superhost', 'has_availability'], inplace = True)


In [15]:
# finding missing values
df_selected.isnull().sum()

id                             0
listing_url                    0
name                           0
picture_url                    0
host_since                     0
host_is_superhost              0
neighbourhood_cleansed         0
latitude                       0
longitude                      0
property_type                  0
accommodates                   0
bedrooms                       1
amenities                      0
price                          0
minimum_nights                 0
maximum_nights                 0
has_availability               0
availability_30                0
availability_60                0
availability_90                0
availability_365               0
number_of_reviews              0
number_of_reviews_ltm          0
number_of_reviews_l30d         0
first_review                   0
last_review                    0
review_scores_rating           0
review_scores_accuracy         4
review_scores_cleanliness      5
review_scores_checkin          4
review_sco

##### From the missing values that are left, we can again see that not many rows are missing. For now I am going to keep it as it is, as I am not sure yet I will use those colums for the data analysis. The 1 missing row from 'bedrooms' I will remove.

In [16]:
df_selected.dropna(subset = ['bedrooms'], inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\3241043554.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.dropna(subset = ['bedrooms'], inplace = True)


In [17]:
# finding missing values
df_selected.isnull().sum()

id                             0
listing_url                    0
name                           0
picture_url                    0
host_since                     0
host_is_superhost              0
neighbourhood_cleansed         0
latitude                       0
longitude                      0
property_type                  0
accommodates                   0
bedrooms                       0
amenities                      0
price                          0
minimum_nights                 0
maximum_nights                 0
has_availability               0
availability_30                0
availability_60                0
availability_90                0
availability_365               0
number_of_reviews              0
number_of_reviews_ltm          0
number_of_reviews_l30d         0
first_review                   0
last_review                    0
review_scores_rating           0
review_scores_accuracy         4
review_scores_cleanliness      5
review_scores_checkin          4
review_sco

In [19]:
# find duplicates
df_dups = df_selected[df_selected.duplicated()]

In [20]:
df_dups

Empty DataFrame
Columns: [id, listing_url, name, picture_url, host_since, host_is_superhost, neighbourhood_cleansed, latitude, longitude, property_type, accommodates, bedrooms, amenities, price, minimum_nights, maximum_nights, has_availability, availability_30, availability_60, availability_90, availability_365, number_of_reviews, number_of_reviews_ltm, number_of_reviews_l30d, first_review, last_review, review_scores_rating, review_scores_accuracy, review_scores_cleanliness, review_scores_checkin, review_scores_communication, review_scores_location, review_scores_value, reviews_per_month]
Index: []

[0 rows x 34 columns]

##### There are no duplicates found.

### 3. Renaming columns

In [21]:
# rename column 'neighbourhood_cleansed':
df_selected.rename(columns = {'neighbourhood_cleansed' : 'location'}, inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\3938977796.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.rename(columns = {'neighbourhood_cleansed' : 'location'}, inplace = True)


In [22]:
df_selected.columns

Index(['id', 'listing_url', 'name', 'picture_url', 'host_since',
       'host_is_superhost', 'location', 'latitude', 'longitude',
       'property_type', 'accommodates', 'bedrooms', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'has_availability',
       'availability_30', 'availability_60', 'availability_90',
       'availability_365', 'number_of_reviews', 'number_of_reviews_ltm',
       'number_of_reviews_l30d', 'first_review', 'last_review',
       'review_scores_rating', 'review_scores_accuracy',
       'review_scores_cleanliness', 'review_scores_checkin',
       'review_scores_communication', 'review_scores_location',
       'review_scores_value', 'reviews_per_month'],
      dtype='object')

In [26]:
# rename column 'number_of_reviews_ltm':
df_selected.rename(columns = {'number_of_reviews_ltm' : 'nr_of_reviews_last12months'}, inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\2338150135.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.rename(columns = {'number_of_reviews_ltm' : 'nr_of_reviews_last12months'}, inplace = True)


In [27]:
df_selected.columns

Index(['id', 'listing_url', 'name', 'picture_url', 'host_since',
       'host_is_superhost', 'location', 'latitude', 'longitude',
       'property_type', 'accommodates', 'bedrooms', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'has_availability',
       'availability_30', 'availability_60', 'availability_90',
       'availability_365', 'number_of_reviews', 'nr_of_reviews_last12months',
       'number_of_reviews_l30d', 'first_review', 'last_review',
       'review_scores_rating', 'review_scores_accuracy',
       'review_scores_cleanliness', 'review_scores_checkin',
       'review_scores_communication', 'review_scores_location',
       'review_scores_value', 'reviews_per_month'],
      dtype='object')

In [28]:
# rename column 'number_of_reviews_l30d':
df_selected.rename(columns = {'number_of_reviews_l30d' : 'nr_of_reviews_last30days'}, inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\953467020.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.rename(columns = {'number_of_reviews_l30d' : 'nr_of_reviews_last30days'}, inplace = True)


In [29]:
df_selected.columns

Index(['id', 'listing_url', 'name', 'picture_url', 'host_since',
       'host_is_superhost', 'location', 'latitude', 'longitude',
       'property_type', 'accommodates', 'bedrooms', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'has_availability',
       'availability_30', 'availability_60', 'availability_90',
       'availability_365', 'number_of_reviews', 'nr_of_reviews_last12months',
       'nr_of_reviews_last30days', 'first_review', 'last_review',
       'review_scores_rating', 'review_scores_accuracy',
       'review_scores_cleanliness', 'review_scores_checkin',
       'review_scores_communication', 'review_scores_location',
       'review_scores_value', 'reviews_per_month'],
      dtype='object')

In [30]:
# rename column 'accommodates':
df_selected.rename(columns = {'accommodates' : 'nr_of_guests'}, inplace = True)

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\1624335910.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected.rename(columns = {'accommodates' : 'nr_of_guests'}, inplace = True)


In [31]:
df_selected.columns

Index(['id', 'listing_url', 'name', 'picture_url', 'host_since',
       'host_is_superhost', 'location', 'latitude', 'longitude',
       'property_type', 'nr_of_guests', 'bedrooms', 'amenities', 'price',
       'minimum_nights', 'maximum_nights', 'has_availability',
       'availability_30', 'availability_60', 'availability_90',
       'availability_365', 'number_of_reviews', 'nr_of_reviews_last12months',
       'nr_of_reviews_last30days', 'first_review', 'last_review',
       'review_scores_rating', 'review_scores_accuracy',
       'review_scores_cleanliness', 'review_scores_checkin',
       'review_scores_communication', 'review_scores_location',
       'review_scores_value', 'reviews_per_month'],
      dtype='object')

### 4. Data Understanding

In [45]:
# basic descriptive statistical analysis 1st half dataset:
df_selected[['id', 'host_since', 'host_is_superhost', 'location', 'latitude', 'longitude', 'nr_of_guests', 'bedrooms', 'price', 'minimum_nights', 'maximum_nights', 'has_availability', 'availability_30', 'availability_60', 'availability_90']].describe()

id      latitude     longitude  nr_of_guests      bedrooms  \
count  1.197700e+04  11977.000000  11977.000000  11977.000000  11977.000000   
mean   1.929839e+17     39.655059      2.993238      5.490273      2.764382   
std    3.450240e+17      0.171639      0.245533      2.623043      1.431615   
min    6.999800e+04     39.302070      2.347260      1.000000      0.000000   
25%    1.972621e+07     39.535645      2.788140      4.000000      2.000000   
50%    3.991464e+07     39.667730      3.050650      6.000000      3.000000   
75%    5.423589e+07     39.802620      3.144890      7.000000      4.000000   
max    1.112024e+18     39.929140      3.474510     16.000000     16.000000   

       minimum_nights  maximum_nights  availability_30  availability_60  \
count    11977.000000    11977.000000     11977.000000     11977.000000   
mean         3.897470      709.190448        13.322117        28.578943   
std          7.131214      483.086641        10.355300        19.263999   
min          1.000000        2.000000         0.000000         0.000000   
25%          1.000000      120.000000         4.000000        12.000000   
50%          3.000000     1125.000000        12.000000        26.000000   
75%          5.000000     1125.000000        22.000000        46.000000   
max        365.000000     1125.000000        30.000000        60.000000   

       availability_90  
count     11977.000000  
mean         43.422894  
std          27.488734  
min           0.000000  
25%          20.000000  
50%          40.000000  
75%          67.000000  
max          90.000000

##### In the availibility columns of 30, 60, 90 and 365 days the minimum is 0, this is realistic as there can be listings that are currently not available or have been fully booked for a certain time period. 

In [32]:
# basic descriptive statistical analysis 2nd half dataset:
df_selected[['number_of_reviews', 'nr_of_reviews_last12months', 'nr_of_reviews_last30days', 'first_review', 'last_review', 'review_scores_rating', 'review_scores_accuracy', 'review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 'review_scores_location', 'review_scores_value', 'reviews_per_month']].describe()

number_of_reviews  nr_of_reviews_last12months  \
count       11980.000000                11980.000000   
mean           27.710017                    7.667947   
std            49.533325                   13.333237   
min             1.000000                    0.000000   
25%             4.000000                    1.000000   
50%            11.000000                    4.000000   
75%            31.000000                   10.000000   
max          1386.000000                  660.000000   

       nr_of_reviews_last30days  review_scores_rating  review_scores_accuracy  \
count              11980.000000          11980.000000            11976.000000   
mean                   0.255008              4.706248                4.729353   
std                    0.811674              0.380863                0.379789   
min                    0.000000              1.000000                1.000000   
25%                    0.000000              4.590000                4.640000   
50%                    0.000000              4.820000                4.850000   
75%                    0.000000              5.000000                5.000000   
max                   24.000000              5.000000                5.000000   

       review_scores_cleanliness  review_scores_checkin  \
count               11975.000000           11976.000000   
mean                    4.686499               4.806375   
std                     0.405871               0.346988   
min                     1.000000               1.000000   
25%                     4.550000               4.750000   
50%                     4.810000               4.920000   
75%                     5.000000               5.000000   
max                     5.000000               5.000000   

       review_scores_communication  review_scores_location  \
count                 11976.000000            11976.000000   
mean                      4.775767                4.705875   
std                       0.376877                0.344723   
min                       1.000000                1.000000   
25%                       4.700000                4.580000   
50%                       4.910000                4.800000   
75%                       5.000000                4.970000   
max                       5.000000                5.000000   

       review_scores_value  reviews_per_month  
count         11975.000000       11980.000000  
mean              4.573675           0.616316  
std               0.428231           0.961637  
min               1.000000           0.010000  
25%               4.440000           0.150000  
50%               4.670000           0.350000  
75%               4.840000           0.760000  
max               5.000000          40.680000

##### The max nr of reviews in the last 12 months can't be 660 as there are only 356 days in the year. I will have to analyze this further. The reviews per month are the average number of reviews per month the listing has over the lifetime of the listing. So it is realistic that these numbers can be low.

In [35]:
# set display options
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [36]:
df_selected['nr_of_reviews_last12months'].value_counts(dropna = False)

nr_of_reviews_last12months
1      1814
0      1733
2      1171
3       830
4       771
5       658
6       572
7       502
8       437
9       395
10      308
11      306
12      259
13      234
14      203
15      155
16      154
17      147
18      127
19      108
21       91
22       89
20       86
25       62
23       61
26       58
27       51
24       50
28       40
29       36
32       36
30       35
31       28
33       28
34       28
35       25
36       17
41       17
37       17
39       16
40       16
43       15
42       14
46       11
45       11
44       10
50        8
55        8
38        8
54        7
47        7
52        7
51        7
56        6
60        6
48        6
53        5
68        5
61        4
71        3
67        3
70        3
59        3
49        3
73        2
83        2
75        2
84        2
79        2
100       2
72        2
78        2
58        2
90        2
63        1
82        1
57        1
99        1
345       1
660       1
121       1
1

##### The review numbers 345, 385 and 660 are the unrealistic numbers. 

In [37]:
df_selected['nr_of_reviews_last12months'] = df_selected['nr_of_reviews_last12months'].replace(['345', '660', '385'], 'NaN')

C:\Users\svand\AppData\Local\Temp\ipykernel_10624\3861698386.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_selected['nr_of_reviews_last12months'] = df_selected['nr_of_reviews_last12months'].replace(['345', '660', '385'], 'NaN')


In [38]:
df_selected['nr_of_reviews_last12months'].value_counts(dropna = False)

nr_of_reviews_last12months
1      1814
0      1733
2      1171
3       830
4       771
5       658
6       572
7       502
8       437
9       395
10      308
11      306
12      259
13      234
14      203
15      155
16      154
17      147
18      127
19      108
21       91
22       89
20       86
25       62
23       61
26       58
27       51
24       50
28       40
29       36
32       36
30       35
31       28
33       28
34       28
35       25
36       17
41       17
37       17
39       16
40       16
43       15
42       14
46       11
45       11
44       10
50        8
55        8
38        8
54        7
47        7
52        7
51        7
56        6
60        6
48        6
53        5
68        5
61        4
71        3
67        3
70        3
59        3
49        3
73        2
83        2
75        2
84        2
79        2
100       2
72        2
78        2
58        2
90        2
63        1
82        1
57        1
99        1
345       1
660       1
121       1
1

In [41]:
#drop the values 345, 385 and 660 in the 'nr_of_reviews_last12months' colum:
df_selected = df_selected[(df_selected.nr_of_reviews_last12months != 345) & (df_selected.nr_of_reviews_last12months != 660) & (df_selected.nr_of_reviews_last12months != 385)]

In [42]:
df_selected['nr_of_reviews_last12months'].value_counts(dropna = False)

nr_of_reviews_last12months
1      1814
0      1733
2      1171
3       830
4       771
5       658
6       572
7       502
8       437
9       395
10      308
11      306
12      259
13      234
14      203
15      155
16      154
17      147
18      127
19      108
21       91
22       89
20       86
25       62
23       61
26       58
27       51
24       50
28       40
29       36
32       36
30       35
33       28
31       28
34       28
35       25
41       17
37       17
36       17
40       16
39       16
43       15
42       14
45       11
46       11
44       10
55        8
50        8
38        8
47        7
52        7
51        7
54        7
56        6
48        6
60        6
68        5
53        5
61        4
67        3
71        3
49        3
70        3
59        3
75        2
83        2
100       2
73        2
72        2
58        2
78        2
90        2
79        2
84        2
244       1
82        1
120       1
99        1
122       1
62        1
121       1
6

In [43]:
df_selected.shape

(11977, 34)

##### Previously there were 11980 rows so the 3 values have been removed succefully.

In [44]:
# export dataframe:
df_selected.to_csv(os.path.join(path, '02 Data','Prepared Data', 'listings_checked.csv'))

In [47]:
# the basic descriptive statistical analysis for the data profile document:
df_selected[['availability_365', 'number_of_reviews', 'nr_of_reviews_last12months', 'nr_of_reviews_last30days', 'first_review', 'last_review', 'review_scores_rating', 'review_scores_accuracy']].describe()

availability_365  number_of_reviews  nr_of_reviews_last12months  \
count      11977.000000       11977.000000                11977.000000   
mean         190.708525          27.418886                    7.553811   
std          100.476925          45.935961                   10.994714   
min            0.000000           1.000000                    0.000000   
25%          109.000000           4.000000                    1.000000   
50%          188.000000          11.000000                    4.000000   
75%          267.000000          31.000000                   10.000000   
max          365.000000         781.000000                  244.000000   

       nr_of_reviews_last30days  review_scores_rating  review_scores_accuracy  
count              11977.000000          11977.000000            11973.000000  
mean                   0.251482              4.706286                4.729389  
std                    0.763245              0.380891                0.379819  
min                    0.000000              1.000000                1.000000  
25%                    0.000000              4.590000                4.640000  
50%                    0.000000              4.820000                4.850000  
75%                    0.000000              5.000000                5.000000  
max                   15.000000              5.000000                5.000000

In [48]:
# the basic descriptive statistical analysis for the data profile document:
df_selected[['review_scores_cleanliness', 'review_scores_checkin', 'review_scores_communication', 'review_scores_location', 'review_scores_value', 'reviews_per_month']].describe()

review_scores_cleanliness  review_scores_checkin  \
count               11972.000000           11973.000000   
mean                    4.686537               4.806404   
std                     0.405909               0.347013   
min                     1.000000               1.000000   
25%                     4.550000               4.750000   
50%                     4.810000               4.920000   
75%                     5.000000               5.000000   
max                     5.000000               5.000000   

       review_scores_communication  review_scores_location  \
count                 11973.000000            11973.000000   
mean                      4.775791                4.705900   
std                       0.376915                0.344746   
min                       1.000000                1.000000   
25%                       4.700000                4.580000   
50%                       4.910000                4.800000   
75%                       5.000000                4.970000   
max                       5.000000                5.000000   

       review_scores_value  reviews_per_month  
count         11972.000000       11977.000000  
mean              4.573688           0.608264  
std               0.428275           0.808354  
min               1.000000           0.010000  
25%               4.440000           0.150000  
50%               4.670000           0.350000  
75%               4.840000           0.760000  
max               5.000000          20.170000